tests/flight/mod.rs

```diff
@@ -33,19 +35,28 @@ use seafowl::frontend::flight::run_flight_server;
 
 mod client;
 mod e2e;
+mod inline_metastore;
 mod search_path;
 mod sync;
 mod sync_fail;
 
-async fn make_test_context() -> Arc<SeafowlContext> {
+async fn make_test_context(local_store: bool) -> Arc<SeafowlContext> {
     // let OS choose a free port
     let listener = TcpListener::bind("127.0.0.1:0").await.unwrap();
     let addr = listener.local_addr().unwrap();
 
+    let object_store_section = if local_store {
+        r#"[object_store]
+type = "local"
+data_dir = "tests/data""#
+    } else {
+        r#"[object_store]
+type = "memory""#
+    };
+
     let config_text = format!(
         r#"
-[object_store]
-type = "memory"
+{object_store_section}
 
 [catalog]
 type = "sqlite"
```

Comment:
Probably the only thing left in this PR is making this catalog optional (expect the metastore to be provided inline if it isn't specified) and erroring out if the user tries to run DDL (creating/deleting tables), which we might be getting automatically because of the `not_impl()` thing

Reply:
How about making it optional and default to in-mem sqlite? 

One can still provide the inline metastores, but we also support the full DDL spectrum, albeit ephemerally.

Reply:
It's possible, but kind of awkward? Since if you first create a table with the inline metastore in use (e.g. use some CREATE TABLE AS statement), you wouldn't be able to query it unless you stopped providing the inline metastore, so I don't yet see how useful that would be? It's the responsibility of the caller to be the metastore/catalog if the inline metastore is being used.

Reply:
Hmm, well you can't create a table with the inline metastore at all, only reads are supported. This is because we implement only the basic `SchemaStore` for the `MemoryStore`, so we the table creation metastore calls would error out (due to `not_impl`).

Reply:
Oh I see, you meant just defaulting to the in-memory metastore if the inline metastore isn't provided. I don't really have strong feelings about it. I guess in the Seafowl configuration where you're using it with the inline metastore it'd surface errors to the developer faster if they forgot to provide it on some code path instead of quietly using the in-mem SQLite on would be my only argument for erroring out.